datalad/core/distributed/clone.py

```diff
@@ -379,11 +380,23 @@ def __call__(
             # Modify .gitmodules to contain originally given url. This is
             # particularly relevant for postclone routines on a later `get`
             # for that subdataset. See gh-5256.
+
+            if isinstance(RI(source), PathRI):
+                # ensure posix paths; Windows paths would neither be meaningful
+                # as a committed path nor are they currently stored correctly
+                # (see gh-7182).
+                # Restricted to when 'source' is identified as a path, b/c this
+                # wouldn't work with file-URLs (ria or not):
+                #
+                # PureWindowsPath("file:///C:/somewhere/path").as_posix() ->
+                # 'file:/C:/somewhere/path'
+                source = PurePath(source).as_posix()
```

Comment:
stripping of leading `./` shouldn't have negative effect for us/git anyhow, should it?

```
In [2]: PurePath('./123').as_posix()
Out[2]: '123'
```

Reply:
Don't think so. That would otherwise bite us everywhere we use `Path`:

```
In [2]: Path('./some')
Out[2]: PosixPath('some')

```

Reply:
Plus: git remains technically irrelevant. This is the input for `datalad get`'s internal `datalad clone` call.